client/finality-grandpa/src/finality_proof.rs

```diff
@@ -338,10 +337,12 @@ pub(crate) fn prove_finality<Block: BlockT<Hash=H256>, B: BlockchainBackend<Bloc
 			if justifies_end_block || justifies_authority_set_change {
 				// check if the proof is generated by the requested authority set
 				if finality_proof.is_empty() {
+					let parent_id = BlockId::Number(current_number - One::one());	
+					let prev_authorities = authorities_provider.authorities(&parent_id)?;
```

Comment:
Couldn't you just keep a reference to `current_authorities` above before reassigning instead of doing another query to the `authorities_provider`? Or just reassign `current_authorities` (and perhaps `authorities_set_id`) below this?